torch/sparse/semi_structured.py

```diff
@@ -232,6 +232,33 @@ def __repr__(self) -> str:  # type: ignore[override]
 
     __torch_function__ = torch._C._disabled_torch_function_impl
 
+    def _pad_tensor_for_matmul(self, original_tensor : torch.Tensor) -> torch.Tensor:
+        """
+        Calculates padding for dense tensor and pads tensor if necessary.
+        If padding is not required, this function returns the original tensor.
+        """
+        # only 2d matmul
+        assert original_tensor.dim() == 2
+
+        # check shape
+        m, n = original_tensor.shape
+        min_rows = _DTYPE_TO_SEMI_STRUCTURED_SPARSE_CONFIG[original_tensor.dtype].min_rows
+        min_cols = _DTYPE_TO_SEMI_STRUCTURED_SPARSE_CONFIG[original_tensor.dtype].min_cols
+        to_pad_m = -m % min_rows if m < min_rows or m % min_rows else 0
+        to_pad_n = -n % min_cols if n < min_cols or n % min_rows else 0
+        if to_pad_m or to_pad_n:
+            warnings.warn(
```

Comment:
Will this warn on every invocation?

Reply:
Yes, we can change this to on first invocation.

Reply:
cc @cpuhrsch 

Actually first invocation might be tricky because this is really a per-matmul warning. So we can't have a warning_shown class variable flag.

And since we create a new instance of the subclass we cannot store if the warning is shown in an instance variable. I will probably add a flag to disable all padding warnings instead. Perhaps we should make a `padding_mode` flag that needs to be set before we automatically do padding.

Reply:
I'd say we remove the warning for now. If someone can't use the 2:4 format because their inputs don't align they have to pad. If it's slower because of padding, then they'll need to adjust inputs. But at least they can use it and try.


But a warning on every invocation can cause a lot of issues and slowdowns (I think even graph breaks maybe?)